src/components/vm/consoles/vnc.jsx

```diff
@@ -48,6 +55,79 @@ const Enum = {
     KEY_DELETE: 111,
 };
 
+export const VncState = ({ vm, vnc }) => {
+    const Dialogs = useDialogs();
+
+    function add_vnc() {
+        Dialogs.show(<AddVNC idPrefix="add-vnc" vm={vm} />);
+    }
+
+    function edit_vnc() {
+        Dialogs.show(<EditVNCModal idPrefix="edit-vnc" vm={vm} consoleDetail={vnc} />);
+    }
+
+    if (vm.state == "running" && !vnc) {
+        return (
+            <EmptyState>
+                <EmptyStateBody>
+                    {_("Graphical support not enabled.")}
+                </EmptyStateBody>
+                <EmptyStateFooter>
+                    <Button variant="secondary" onClick={add_vnc}>
+                        {_("Add VNC")}
+                    </Button>
+                </EmptyStateFooter>
+            </EmptyState>
+        );
+    }
+
+    let vnc_info;
+    let vnc_action;
+
+    if (!vnc) {
+        vnc_info = _("not supported");
+        vnc_action = (
+            <Button variant="link" isInline onClick={add_vnc}>
+                {_("Add VNC")}
+            </Button>
+        );
+    } else {
+        if (vnc.port == -1)
+            vnc_info = _("VNC, dynamic port");
```

Comment:
With a dynamic port it would be super helpful to actually *show* it, so that you know what to type into your virt-viewer. Can we get this from libvirt somehow? Is the port part of the downloaded `.vv` file?

Reply:
Yes. You can see the actual port in the "Desktop viewer" tab of a running machine.